### PR TITLE
Finish support for AD time derivatives of dofs

### LINKED ISSUE #28859: Central time integrator lags derivatives for AD nodal values
### Bug Description

It seems when using the adCoupledDot on nodal variables on a nonlinear variable, to set an axuvariable, the values are lagged for AD compared to non AD

### Steps to Reproduce

Run the coupled_dots_nodal.i input from https://github.com/idaholab/moose/pull/28857/files with a CentralDifference time integrator

### Impact

Wrong values lead to wrong results

### [Optional] Diagnostics

_No response_

### PR DESCRIPTION
**Pull request by GiudGiud**
also includes a bug fix for second order AD dot dots
closes #28859
refs #28867

### CODE REVIEW

File: framework/src/variables/MooseVariableData.C [hunk 1]
```diff
@@ -1794,6 +1803,24 @@ MooseVariableData<OutputType>::fetchADDoFValues()
...
+                                                   _need_ad_u_dotdot ? _ad_dofs_dotdot[i]
+                                                                     : _ad_real_dummy);
+      }
+      // Executing something with a time derivative at initial should not put a NaN
+      else if (_time_integrator && !_time_integrator->dt())
+        _ad_dofs_dot[i] = 0;
+      else
+        mooseError("AD nodal time derivatives not implemented for variables without a time "
+                   "integrator (auxiliary variables)");
+    }
```
Comment (lindsayad) [inline framework/src/variables/MooseVariableData.C:1823]: `computeNodalValues` is the only caller of `fetchADDoFValues`. I've always thought we should sync the names better. As it is, I think it's a bit confusing

File: framework/include/variables/MooseVariableData.h [hunk 2]
```diff
@@ -726,6 +731,17 @@ MooseVariableData<OutputType>::adDofValues() const
   return _ad_dof_values;
 }
 
+template <typename OutputType>
+const MooseArray<ADReal> &
+MooseVariableData<OutputType>::adDofValuesDot() const
+{
+  _need_ad = _need_ad_u_dot = true;
```
[the remark this answers is not in the document]
Comment (GiudGiud) [reply framework/include/variables/MooseVariableData.h:738]: that's true but I m following the rest of the code in that regard, re-using that same boolean

File: framework/src/variables/MooseVariableData.C (hunk 1 above)
Comment (GiudGiud) [reply framework/src/variables/MooseVariableData.C:1823]: ok I ll rename!

File: test/src/auxkernels/CoupledAux.C [hunk 3]
```diff
@@ -10,9 +10,11 @@
 #include "CoupledAux.h"
 
 registerMooseObject("MooseTestApp", CoupledAux);
+registerMooseObject("MooseTestApp", ADCoupledAux);
 
+template <bool is_ad>
 InputParameters
-CoupledAux::validParams()
+CoupledAuxTempl<is_ad>::validParams()
```
Comment (lindsayad) [inline test/src/auxkernels/CoupledAux.C:17]: It seems like these changes are really only for testing purposes? Is there any reason in production to use an `ADCoupledAux`?

Comment (lindsayad) [reply test/src/auxkernels/CoupledAux.C:17]: And I see that you are modifying a test material ... so do we really need these aux kernel changes?

Comment (GiudGiud) [reply test/src/auxkernels/CoupledAux.C:17]: I thought the changes didn't overlap but they might have ended up overlapping. I ll check 

No reason to use ADcoupledAux in production. If they don't overlap do you prefer creating a new object in test/ over templating in the framework ?

Comment (lindsayad) [reply test/src/auxkernels/CoupledAux.C:17]: I would prefer that yea. Then we don't need a bunch of `raw_value` calls

**Review by lindsayad — APPROVED** [review]

**Comment by GiudGiud** [comment]
OpenMPI failure unrelated (previous runs have different failures)
Valgrind failures from previous run were unrelated (timeouts)

**Comment by GiudGiud** [comment]
Thanks for the review!

**Pull request merged**